worker/wallet.js

```diff
@@ -255,14 +255,34 @@ async function checkWithdrawal ({ data: { hash }, boss, models, lnd }) {
   }
 }
 
-export async function autoDropBolt11s ({ models }) {
-  await serialize(models, models.$executeRaw`
-    UPDATE "Withdrawl"
-    SET hash = NULL, bolt11 = NULL
+export async function autoDropBolt11s ({ models, lnd }) {
+  const invoices = await models.$queryRaw`
+    SELECT id, hash, bolt11 FROM "Withdrawl"
     WHERE "userId" IN (SELECT id FROM users WHERE "autoDropBolt11s")
     AND now() > created_at + interval '${INVOICE_RETENTION_DAYS} days'
-    AND hash IS NOT NULL;`
-  )
+    AND hash IS NOT NULL`
+
+  const successfulDeletes = []
+  const failedDeletes = []
+  for (const invoice of invoices) {
+    try {
+      await deletePayment({ id: invoice.hash, lnd })
+      successfulDeletes.push(invoice)
+    } catch (error) {
+      console.error(`Error removing invoice with hash ${invoice.hash}:`, error)
+      failedDeletes.push(invoice)
+    }
+  }
+  if (successfulDeletes.length > 0) {
+    await models.withdrawl.deleteMany({
+      where: { hash: { in: successfulDeletes } }
+    })
+  }
+  if (failedDeletes.length > 0) {
+    await models.withdrawl.updateMany({
+      where: { hash: { in: failedDeletes } }, data: { status: 'UNKNOWN_FAILURE' }
```

Comment:
You shouldn't overwrite the `status` column. That's to indicate the status of the withdrawal itself (`ROUTE_NOT_FOUND`, `CONFIRMED` etc.), not the status of a delete.

The `UPDATE` query in the existing code was good as it was.

Reply:
Ok I reverted it to setting those values to null and then adding the ones back into the database if they failed. Should I add another column to the withdrawl model to show an unsuccessful delete from LND?

Reply:
> Should I add another column to the withdrawl model to show an unsuccessful delete from LND?

No, I think it's fine to just log the errors and keep retrying. At some point, we should log such errors into some channel that we monitor.